### PR TITLE
add initial rulesets

### PR DESCRIPTION
**Pull request by pq**
Content carried over from WIP in: https://github.com/dart-lang/linter/tree/master/tool/canonical

@munificent: could you spot check?

/cc @devoncarew @mit-mit @goderbauer

### CODE REVIEW

File: lib/core.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+linter:
```
Comment (devoncarew) [inline lib/core.yaml:1]: As a drive by comment, we should have yaml comments here (and in `recommend.yaml`) describing what the files are, how to use them, where the format docs are, ...

We could do that as part of this PR (or, track the work in an issue in this repo).

Comment (pq) [reply lib/core.yaml:1]: It's a good point.  I think we want some thoughtful docs in general (here, README, etc).  Maybe we can land this (and #1) and iterate so we can start experimenting?

Tracking issue for docs: https://github.com/dart-lang/lints/issues/5

**Review by devoncarew — APPROVED** [review]

**Comment by parlough** [comment]
Just as a question, once these are released, do you think you'll include them in the sets fields of the machine generated JSON?

**Comment by pq** [comment]
@parlough: totally, we should do that.  could you maybe open an issue on the linter to track?

**Comment by parlough** [comment]
> @parlough: totally, we should do that. could you maybe open an issue on the linter to track?

For referenced, open an issue at https://github.com/dart-lang/linter/issues/2611

File: lib/recommend.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+linter:
```
Comment (goderbauer) [inline lib/recommend.yaml:1]: To echo @devoncarew's comment above: This file should probably contain a comment explaining that it should be used in addition to core.yaml - and not as a substitute.

(That was my first question I had first time looking at these files)

File: lib/core.yaml (hunk 1 above)
Comment (mit-mit) [inline lib/core.yaml:1]: I wonder if we want the file path here to be `/lib/dart/core.yaml` so that we could also have a `/lib/flutter/core.yaml` ?

Comment (pq) [reply lib/core.yaml:1]: Interesting.  Scoping has some other nice benefits too.

@goderbauer @devoncarew : wdyt?

Comment (devoncarew) [reply lib/core.yaml:1]: TBH I think that supporting frameworks in this package will be challenging. Once we add one there's no reason we shouldn't then support more; changes here will be tightly versioned together even though the frameworks will be logically unrelated.

I'd optimize for the 'dart' case here, and just have the two main files in the top level of the `lib/` directory.

Comment (pq) [reply lib/core.yaml:1]: > I'd optimize for the 'dart' case here,

I guess if we leave it as is, we could always add directories for frameworks (if we added them).  So we might have something like:

    lib/
     core.yaml
     recommend.yaml
     framework_1/
       core.yaml
       recommend.yaml
     framework_2/
       core.yaml
     ...


What do we think about `flutter` in the short term?  Top-level, nested or somewhere else?

Comment (goderbauer) [reply lib/core.yaml:1]: FWIW, I don't think we will have multiple lint sets for flutter. There'll just be a `flutter.yaml` set,

File: lib/recommend.yaml (hunk 2 above)
Comment (devoncarew) [inline lib/recommend.yaml:1]: I believe this file should have an `include: package:lints/core.yaml` line.

Comment (goderbauer) [reply lib/recommend.yaml:3]: Regarding https://github.com/dart-lang/lints/pull/9#discussion_r621696293, should this then import core.yaml and not be used in combination with the other file?

Comment (pq) [reply lib/recommend.yaml:3]: Yep.  Done. 👍

**Comment by pq** [comment]
Landing and we can iterate.

Thanks for all the thoughtful back and forth!

**Pull request merged**